### ISSUE TITLE
Unable to disable warning when using derive

### ISSUE THREAD

**Issue by alexkazik** [rust-lang/rust-clippy#9179]
### Summary

I want to silence a clippy warning.

I did stick the allow to all places I thought possible.

It is silenced without the derive.

### Lint Name

type_complexity

### Reproducer

I tried this code:

```rust
use std::borrow::Cow;

#[derive(serde::Serialize, serde::Deserialize)]
pub enum EncounterType {}

#[allow(clippy::type_complexity)]
#[derive(serde::Serialize, serde::Deserialize)]
#[allow(clippy::type_complexity)]
pub struct SerdeEncounter<'a>(
    #[allow(clippy::type_complexity)] Vec<(Cow<'a, str>, Vec<EncounterType>, usize, Cow<'a, str>)>,
);
```

I saw this happen:

```
warning: very complex type used. Consider factoring parts into `type` definitions
  --> app/src/lib.rs:99:39
   |
99 |     #[allow(clippy::type_complexity)] Vec<(Cow<'a, str>, Vec<EncounterType>, usize, Cow<'a, str>)>,
   |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity
```

I expected to see this happen:
_nothing_

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: aarch64-apple-darwin
release: 1.62.0
LLVM version: 14.0.5
```

**Comment by taquangtrung** [comment]
I also encountered the same error when clippy reported errors on the auto-generated code by the [Pest parser generator](https://github.com/pest-parser/pest).

Here is my code: 
```rust
/// Data structure representing a parse tree of Yul intermediate code.
///
/// This data structure is automatically derived by the `Pest` parser generator.
#[derive(Parser)]
#[grammar = "yul.pest"]
struct YulParser;
```

and the lining rules:
```rust
#![cfg_attr(feature = "linting", deny(clippy::suspicious))]
```

And below is the error reported by Clippy.  How is it possible to disable Clippy from checking linting rules for the code generated by `#[derive(Parser)]`?

```
cargo +stable clippy --tests --benches --features linting
    Checking yulc v0.1.0 (/home/trungtq/workspace/yulc)
error: almost complete ascii letter range
  --> yulc/src/parser.rs:16:10
   |
16 | #[derive(Parser)]
   |          ^^^^^^
   |
note: the lint level is defined here
  --> yulc/src/lib.rs:15:39
   |
15 | #![cfg_attr(feature = "linting", deny(clippy::suspicious))]
   |                                       ^^^^^^^^^^^^^^^^^^
   = note: `#[deny(clippy::almost_complete_letter_range)]` implied by `#[deny(clippy::suspicious)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#almost_complete_letter_range
   = note: this error originates in the derive macro `Parser` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by giraffate** [comment]
This was fixed by https://github.com/rust-lang/rust-clippy/pull/9467, so I'm closing this.

**Comment by alexkazik** [comment]
At lest clippy 0.1.65 (2022-09-26) does not fix my issue:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=779725236b9bfeccf85248c64f8acd49
